src/app/state.service.ts

```diff
@@ -82,6 +82,14 @@ export class StateService {
     this.dispatchAction(ActionTypes.NEXT_EXERCISE);
   }
 
+  toggleAutorun() {
+    this.dispatchAction(ActionTypes.TOGGLE_AUTORUN);
+  }
+
+  getAutorun() {
```

Comment:
@kirjs , I removed it. It seemed, initially `state.service` to be the best place to maintain state for that but it's behind by a `onCodeChange` cycle. Currently, exercise.component maintains that autorun state. This was the was driving reason to, perhaps, have a simple autoruncontrol.service for that. Thoughts welcome.